cmd/src/campaigns_common.go

```diff
@@ -99,6 +104,21 @@ func campaignsDefaultCacheDir() string {
 	return path.Join(uc, "sourcegraph", "campaigns")
 }
 
+// campaignsDefaultTempDirPrefix returns the prefix to be passed to ioutil.TempFile. If the
+// environment variable SRC_CAMPAIGNS_TMP_DIR is set, that is used as the
+// prefix. Otherwise we use "/tmp".
+func campaignsDefaultTempDirPrefix() string {
+	p := os.Getenv("SRC_CAMPAIGNS_TMP_DIR")
+	if p != "" {
+		return p
+	}
+	// We use an explicit prefix for our temp directories, because otherwise Go
+	// would use $TMPDIR, which is set to `/var/folders` per default on macOS. But
+	// Docker for Mac doesn't have `/var/folders` in its default set of shared
+	// folders, but it does have `/tmp` in there.
+	return "/tmp"
```

Comment:
I know this is replicating the existing behaviour, but I wonder if we should actually use [`os.TempDir()`](https://golang.org/pkg/os/#TempDir) on non-macOS systems. Probably something we can do as a follow up, though.

Reply:
Done! Added the explicit "if macOS use /tmp" check and fallback to `os.TempDir()`.